test/extensions/filters/common/ext_authz/ext_authz_http_impl_test.cc

```diff
@@ -513,6 +514,38 @@ TEST_F(ExtAuthzHttpClientTest, AuthorizationDeniedWithAllAttributes) {
                      TestCommon::makeMessageResponse(expected_headers, expected_body));
 }
 
+// Verify client response status code and body when runtime guard for ext_authz_http_service_5xx_is_denied_instead_of_error is false
+TEST_F(ExtAuthzHttpClientTest, AuthorizationRequestErrorWithRuntimeGuardFalse) {
+  TestScopedRuntime scoped_runtime;
+  scoped_runtime.mergeValues(
+      {{"envoy.reloadable_features.ext_authz_http_service_5xx_is_denied_instead_of_error",
+        "false"}});
+
+  const std::string yaml = R"EOF(
+  http_service:
+    server_uri:
+      uri: "ext_authz:9000"
+      cluster: "ext_authz"
+      timeout: 0.25s
+  failure_mode_allow: true
+  )EOF";
+
+  initialize(yaml);
```

Comment:
We don't need this?

Tips: what we need is to "fix" `ExtAuthzHttpClientTest.AuthorizationRequest5xxError` test by setting the corresponding runtime flag to false (and renaming it with e.g. `ExtAuthzHttpClientTest.AuthorizationRequest5xxErrorWhenRuntimeFlagFalse`). Then adapt the `ExtAuthzHttpClientTest.AuthorizationRequest5xxError`.


Something like the following:

```cpp
// Test the client when a call to authorization server returns a 5xx error status.
TEST_F(ExtAuthzHttpClientTest, AuthorizationRequest5xxError) {
  const auto expected_body = std::string{"test"};
  const auto expected_headers = TestCommon::makeHeaderValueOption(
      {{":status", "500", false}, {"foo", "bar", false}, {"x-foobar", "bar", false}});
  const auto authz_response = TestCommon::makeAuthzResponse(
      CheckStatus::Denied, Http::Code::InternalServerError, expected_body, expected_headers);

  envoy::service::auth::v3::CheckRequest request;
  client_->check(request_callbacks_, request, parent_span_, stream_info_);

  EXPECT_CALL(request_callbacks_,
              onComplete_(WhenDynamicCastTo<ResponsePtr&>(AuthzDeniedResponse(authz_response))));
  client_->onSuccess(async_request_,
                     TestCommon::makeMessageResponse(expected_headers, expected_body));
}
```

So now when the authorization error sends 5xx, with headers and body, we expect to receive: `AuthzDeniedResponse(authz_response)`.

Reply:
Please CMIIW mas. but I think we don't need to fix the existing `ExtAuthzHttpClientTest.AuthorizationRequest5xxError` test because the runtime guard is true by default, right? So this test will ensure that the existing behavior is not changed.

Thus, we only need to add a new test `ExtAuthzHttpClientTest.AuthorizationRequest5xxErrorWhenRuntimeFlagFalse` to check new behavior and expect `AuthzDeniedResponse(authz_response)`.